Source/WebCore/editing/Editing.cpp

```diff
@@ -1499,6 +1502,83 @@ SimpleRange adjustToVisuallyContiguousRange(const SimpleRange& range)
     }, SelectionExtentMovement::Closest).value_or(range);
 }
 
+EnclosingLayerInfomation computeEnclosingLayer(bool isContentEditable, const SimpleRange& range)
+{
+    VisibleSelection selection = VisibleSelection(range);
+
+    if (selection.isNoneOrOrphaned())
+        return { nullptr, nullptr, nullptr, { } };
+
+    if (!isContentEditable && selection.isCaret())
+        return { nullptr, nullptr, nullptr, { } };
```

Comment:
It's a bit weird to pass in a `range` here that's been converted from a `VisibleSelection`, right before converting the range _back_ into a `VisibleSelection`.

You could consider either changing this to take a `VisibleSelection`, or keep the range and use `positionsForRange` to get the `start` and `end`, and then `isEditablePosition` on either the `start` or `end` to check if it's editable. You'd then change the `selection.isCaret()` check to use `range.collapsed()` instead.